File: docs/static/img/.aws-oxygen/aws-shared/mainline/en_us/code-samples/javav2/example_code/sagemaker/src/main/java/com/example/sage/DeleteModel.java

/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/

package com.example.sage;

import software.amazon.awssdk.regions.Region;
import software.amazon.awssdk.services.sagemaker.model.DeleteModelRequest;
import software.amazon.awssdk.services.sagemaker.SageMakerClient;
import software.amazon.awssdk.services.sagemaker.model.SageMakerException;


/**
 * To run this Java V2 code example, ensure that you have setup your development environment, including your credentials.
 *
 * For information, see this documentation topic:
 *
 * https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/get-started.html
 */
public class DeleteModel {

    public static void main(String[] args) {

        final String USAGE = "\n" +
                "Usage:\n" +
                "    <modelName>\n\n" +
                "Where:\n" +
                "    modelName - The name of the model.\n\n";

        if (args.length != 1) {
            System.out.println(USAGE);
            System.exit(1);
        }

        String modelName = args[0];
        Region region = Region.US_WEST_2;
        SageMakerClient sageMakerClient = SageMakerClient.builder()
                .region(region)
                .build();

        deleteSagemakerModel(sageMakerClient, modelName);
        sageMakerClient.close();
    }

    public static void deleteSagemakerModel(SageMakerClient sageMakerClient, String modelName) {

       try {
            DeleteModelRequest deleteModelRequest = DeleteModelRequest.builder()
                .modelName(modelName)
                .build();

            sageMakerClient.deleteModel(deleteModelRequest);

       } catch (SageMakerException e) {
           System.err.println(e.awsErrorDetails().errorMessage());
           System.exit(1);
       }
    }
}

File: docs/static/img/.aws-oxygen/aws-shared/mainline/en_us/code-samples/kotlin/services/sagemaker/src/main/kotlin/com/kotlin/sage/DeleteModel.kt

/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/

package com.kotlin.sage

import aws.sdk.kotlin.services.sagemaker.SageMakerClient
import aws.sdk.kotlin.services.sagemaker.model.DeleteModelRequest
import kotlin.system.exitProcess

/**
To run this Kotlin code example, ensure that you have setup your development environment,
including your credentials.

For information, see this documentation topic:
https://docs.aws.amazon.com/sdk-for-kotlin/latest/developer-guide/setup.html
 */
suspend fun main(args:Array<String>) {

    val usage = """
    Usage:
        <modelName>

    Where:
        modelName - The name of the model.
    """

    if (args.size != 1) {
        println(usage)
        exitProcess(1)
    }

    val modelName = args[0]
    deleteSagemakerModel(modelName)
    }

suspend fun deleteSagemakerModel(modelNameVal: String?) {

     val request = DeleteModelRequest {
            modelName = modelNameVal
     }

    SageMakerClient { region = "us-west-2" }.use { sageMakerClient ->
        sageMakerClient.deleteModel(request)
    }
}
